code/src/main/kotlin/com/expediagroup/sdk/v2/core/apache/util/RequestConfigUtil.kt

```diff
@@ -5,6 +5,16 @@ import com.expediagroup.sdk.v2.core.trait.configuration.SocketTimeoutTrait
 import org.apache.http.client.config.RequestConfig
 
 
+/**
+ * Creates a copy of the default `RequestConfig` and sets the connection and socket timeouts.
+ *
+ * This method ensures that the provided `configuration` implements the `SocketTimeoutTrait`,
+ * which provides the required `getSocketTimeout` method to retrieve the socket timeout value.
+ *
+ * @param configuration The `ClientConfiguration` instance that must implement `SocketTimeoutTrait`.
+ * @return A configured `RequestConfig` instance with the connection and socket timeouts set, or `null` if the configuration is invalid.
+ * @throws IllegalArgumentException If the provided `ClientConfiguration` does not implement `SocketTimeoutTrait`.
+ */
```

Comment:
We can move this also to the file that builds the http client, it's a simple internal function, I don't think it has to live in a separate file, what do you think?

Reply:
Done.